### PR TITLE
Update secrets docs.

### PR DESCRIPTION
**Pull request by erictune**
Use `kubectl create secret` wherever possible
instead of defining a complete object.

Reorganized some content.

### CODE REVIEW

**Comment by erictune** [comment]
Fixes #18912

**Comment by erictune** [comment]
@janetkuo

**Comment by erictune** [comment]
ping

File: docs/user-guide/secrets.md [hunk 1]
```diff
@@ -90,9 +99,70 @@ this is the recommended workflow.
...
+====
+password.txt:	13 bytes
+username.txt:	6 bytes
+```
+
+Note that neither `get` nor `describe` shows the contents of the file by default.
+This is to protect the secret from being exposed accidentally to someone looking
+or from being stored in a terminal log.
+
+See [#decoding-a-secret] for how to see the contents.
```
Comment (pmorie) [inline docs/user-guide/secrets.md]: This doesn't render as a link in markdown

File: docs/user-guide/secrets.md [hunk 2]
```diff
@@ -101,23 +171,58 @@ metadata:
...
+Decode the password field:
+
+```console
+$ echo "MWYyZDFlMmU2N2RmCg==" | base64 -D
+1f2d1e2e67df
+```
 
 ### Manually specifying a Secret to be Mounted on a Pod
 
+Once You have the secret is created, you can need to modify your pod to specify
```
Comment (pmorie) [inline docs/user-guide/secrets.md]: typo: You

Comment (pmorie) [inline docs/user-guide/secrets.md]: s/You//

Comment (pmorie) [inline docs/user-guide/secrets.md]: Once the secret is created, you can create a pod that consumes that secret.

File: docs/user-guide/secrets.md [hunk 3]
```diff
@@ -158,11 +263,23 @@ whichever is convenient.
 
 See another example of creating a secret and a pod that consumes that secret in a volume [here](secrets/).
 
-### Manually specifying an imagePullSecret
+### Using Secrets
+
+Secrets can be mounted as data volumes to be used by a container in a pod.
```
Comment (pmorie) [inline docs/user-guide/secrets.md]: I'll add doc on secret in envs as a follow-up to this :)

**Comment by pmorie** [comment]
@erictune, a couple minor nits, LGTM otherwise.  I'll do a follow-up docing secrets in env vars after this merges :)

**Comment by pmorie** [comment]
@erictune minor nit, also: no periods in commit messages

**Comment by pmorie** [comment]
LGTM @erictune, thanks a lot

**Pull request merged**